### PR TITLE
Log weather periodically via OpenWeatherAPI - proof of concept/WIP

### PR DESCRIPTION
**Pull request by CloCkWeRX**
#863
For simplicity, I've just left a default/hardcoded API key in so its easy to play with - that'll be gone by the final version

TODO:
 * Check the ODBL license is appropriate to integrate with our DB.
 * Check if its really ODBL or CCBYSA (http://openweathermap.org/price is unclear)
 * Staging, production, etc accounts
 * Frontend display of data
 * API limits.
 * On deployment; configuring https://devcenter.heroku.com/articles/scheduler
 * On deployment, ensure we are on postgres 9.4 or higher (https://devcenter.heroku.com/articles/heroku-postgresql#version-support-and-legacy-infrastructure)

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
Super annoying, I have no idea how likes_controller has changes on it.

File: app/models/planting_weather_log.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#     "message":0.0097,
+#     "country":"GB",
+#     "sunrise":1491801287,
+#     "sunset":1491850179
+#   },
+#   "id":2643743,
+#   "name":"London",
+#   "cod":200
+# }
+class PlantingWeatherLog < ActiveRecord::Base
```
Comment (Br3nda) [inline app/models/planting_weather_log.rb]: Shouldn't this be per garden? Or even per location? The basil and the tomatoes are next to each other so have the same weather afaik

**Comment by Br3nda** [comment]
Btw I have and old fork of grow stuff with moisture , light, temp,and fertiliser readngs per planting (actually per sensor, which is associated with many plantings).

**Comment by CloCkWeRX** [comment]
> schema.rb changing
That's my local DB having a bunch of different feature branches on it; and me (tiredly) doing a rubocop --auto-correct && git commit -a; sigh.

Prior to merge, I'll go clean it up to be only relevant bits.

File: app/models/planting_weather_log.rb (hunk 1 above)
Comment (CloCkWeRX) [inline app/models/planting_weather_log.rb]: I went for planting so you can answer things like "Why did my leeks grow better last year than this year? Oh, 27 days of 40 degree temperature in a row on last year's planting" more easily 

IE its simpler to do ```SELECT AVG(temp) FROM planting_weather_log WHERE planting_id = ?``` and only have data recorded for the duration of the planting; than it is to do

```SELECT AVG(temp) FROM garden_weather_log JOIN gardens JOIN plantings WHERE garden_weather_log.created_at BETWEEN planting.start and planting.end GROUP BY planting_id```
 x lots.


Plantings also have a start and end date usually (planting.active); while a Garden is forever; and at least one per user; even if they new do anything with the site. 
I suspect we'd run out of the free tier of API usage if we did that for everyone's gardens.

For dealing with plantings in the same garden/location/etc; we can probably cache the API requests at a city/geocoded location level.

**Comment by CloCkWeRX** [comment]
Will pick this up again in a while

**Pull request closed without merging**